parsers/GB.py

```diff
@@ -1,71 +1,104 @@
+    url = f"http://eco2mix.rte-france.com/curves/getDonneesMarche?dateDeb={day_start}&dateFin={day_end}&mode=NORM"
 
-    response = r.get(url)
-    obj = ET.fromstring(response.content)
-    datas = {}
+    session = session or Session()
+    response = session.get(url)
+
+    if not response.ok:
+        raise ParserException(
+            PARSER,
+            f"Exception when fetching price error code: {response.status_code}: {response.text}",
+            zone_key,
+        )
 
-    for donnesMarche in obj:
-        if donnesMarche.tag != "donneesMarche":
-            continue
+    xml_tree = ET.fromstring(response.content)
 
-        start_date = arrow.get(
-            arrow.get(donnesMarche.attrib["date"]).datetime, "Europe/Paris"
+    price_list = PriceList(logger=logger)
+    for daily_market_data in xml_tree.iterfind("donneesMarche"):
+        day = datetime.strptime(daily_market_data.get("date"), "%Y-%m-%d").replace(
+            tzinfo=timezone.utc
         )
+        for daily_zone_data in daily_market_data:
+            zone_code = daily_zone_data.get("perimetre")
 
-        for item in donnesMarche:
-            if item.get("granularite") != "Global":
+            # Data for Germany / Luxembourg is not set / reported as aggregate region
+            if zone_code in {"DE", "DL"}:
                 continue
-            country_c = item.get("perimetre")
-            if zone_key != country_c:
+
+            if zone_key != zone_code:
+                continue
+
+            granularity = daily_zone_data.get("granularite")
+            if granularity != "Global":
                 continue
-            value = None
-            for value in item:
-                if value.text == "ND":
+
+            for value in daily_zone_data:
+                price = None if value.text == "ND" else float(value.text)
+                if price is None:
                     continue
-                period = int(value.attrib["periode"])
-                datetime = start_date.shift(hours=+period).datetime
-                if datetime not in datas:
-                    datas[datetime] = {
-                        "zoneKey": zone_key,
-                        "currency": "EUR",
-                        "datetime": datetime,
-                        "source": "rte-france.com",
-                    }
-                data = datas[datetime]
-                data["price"] = float(value.text)
-
-    return list(datas.values())
+
+                period_number = int(value.attrib["periode"])
+                dt = day + timedelta(hours=period_number)
+                is_future = dt > now
+
+                price_list.append(
+                    zoneKey=zone_key,
+                    datetime=dt,
+                    source="rte-france.com",
+                    price=price,
+                    currency="EUR",
+                    sourceType=EventSourceType.forecasted
+                    if is_future
+                    else EventSourceType.measured,
```

Comment:
If the price is set on a day ahead market it is fine to say it's only measured as this won't change.

We should probably add a `EventSourceType.day_ahead`.

Reply:
I see! Thanks @VIKTORVAV99 

I just realised now then that this API can return prices from the full day ahead, just by updating the code to ask for `dateFin` = day + 1 (and to double check data for day + 2 or more returns nulls instead of having values). Is this something that would be useful to add here? (and I imagine still marking them as 'measured' right?)

Reply:
Yes that would be very nice! It would open up more possibilities in the future.

Reply:
Perfect! I have updated to return day-ahead prices for live data, but let me know if you prefer to return "day ahead" prices also for historical data